components/camel-zipkin/src/main/java/org/apache/camel/zipkin/ZipkinTracer.java

```diff
@@ -625,8 +625,15 @@ private Span serverRequest(Tracing brave, String serviceName, Exchange exchange)
             state = new ZipkinState();
             exchange.setProperty(ZipkinState.KEY, state);
         }
-
-        Span span = brave.tracer().nextSpan(EXTRACTOR.extract(exchange.getIn()));
+        
+        Span span = null;
+        if (ObjectHelper.isEmpty(EXTRACTOR.extract(exchange.getIn()))) {
```

Comment:
I wonder if we should store the result of EXTRACTOR in a local variable so we only call this once, as we can use this value in the else block below